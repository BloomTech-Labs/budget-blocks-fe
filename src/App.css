@import '~destyle.css';

* {
  box-sizing: border-box;
  font-family: 'Poppins', sans-serif;
}

html {
  font-size: 62.5%; /* equals to 10 px */
}

.progressBar{
  background: rgba(164, 234, 230, 0.43);
  position: relative;
  height: 20px;
  width: 199px;
  border-radius: 6px;
  border: 1px solid rgba(164, 234, 230, 0.43);
}

.filler{
  height: 100%;
  background: #5ACEC5;
  border-radius: inherit;
  transition: width .2s ease-in;
}