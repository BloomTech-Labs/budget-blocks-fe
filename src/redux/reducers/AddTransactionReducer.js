import {
	ADD_TRANSACTION_LOADING,
	ADD_TRANSACTION_FAILED,
	ADD_TRANSACTION_SUCCESS,
	CATEGORIES_LOADING,
	CATEGORIES_SUCCESS,
	CATEGORIES_FAILED,
	DELETE_TRANS_FAIL,
	DELETE_TRANS_LOADING,
	DELETE_TRANS_SUCCESS
} from '../actions/AddTransactionActions';

export const reducer = (state = initialState, action) => {
	switch (action.type) {
		case ADD_TRANSACTION_LOADING:
			return {
				...state,
				isFetching: true,
				error: null
			};
		case ADD_TRANSACTION_FAILED:
			return {
				...state,
				isFetching: false,
				error: action.payload
			};
		case ADD_TRANSACTION_SUCCESS:
			return {
				...state,
				transaction: action.payload,
				isFetching: false,
				error: null
			};
		///
		case CATEGORIES_LOADING:
			return {
				...state,
				isFetching: true,
				error: null
			};
		case CATEGORIES_FAILED:
			return {
				...state,
				isFetching: false,
				error: action.payload
			};
		case CATEGORIES_SUCCESS:
			return {
				...state,
				categories: action.payload,
				isFetching: false,
				error: null
			};

		//delete trans
		case DELETE_TRANS_LOADING:
			return {
				...state,
				transaction: action.payload,
				isFetching: true,
				error: ''
			};
		case DELETE_TRANS_SUCCESS:
			return {
				...state,
				transaction: action.payload,
				isFetching: false,
				error: ''
			};
		case DELETE_TRANS_FAIL:
			return {
				...state,
				isFetching: false,
				error: `${action.payload.status}`
			};

		default:
			return state;
	}
};

const initialState = {
	categories: '',
	transaction: '',
	error: null,
	isFetching: false
};
