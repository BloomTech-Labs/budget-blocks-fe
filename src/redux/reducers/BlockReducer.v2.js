export const blocksReducer = (state = { blocks: [], blockExpenses: {} }, action) => {
  switch (action.type) {
    case "HYDRATE_BLOCKS":
      return {
        ...state,
        blocks: action.payload.blocks,
        blockExpenses: action.payload.blockExpenses
      }
    case "DELETE_BLOCK":
      return {
        ...state,
        blocks: action.payload
      }
    case "ADD_BLOCK":
      return {
        ...state,
        blocks: action.payload
      }
    case "UPDATE_BLOCKS":
      return {
        ...state,
        blocks: action.payload
      }
    case "HYDRATE_OWNEXPENSES":
      const newBlockExpenses = { ...state.blockExpenses }
      console.log('newBlockExpensesBEFORE', newBlockExpenses, action.payload.blockID, action.payload.expenses)
      newBlockExpenses[action.payload.blockID] = action.payload.expenses
      console.log('newBlockExpensesAFTER', newBlockExpenses, action.payload.blockID, action.payload.expenses)

      return {
        ...state,
        blockExpenses: newBlockExpenses
      }
    default: {
      return state
    }
  }
}