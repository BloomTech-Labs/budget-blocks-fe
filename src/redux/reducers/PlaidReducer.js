
import {
    SEND_LINK_TOKEN_LOADING,
    SEND_LINK_TOKEN_SUCCESS,
    SEND_LINK_TOKEN_FAILED,
    GET_TRANS_LOADING,
    GET_TRANS_SUCCESS,
    GET_TRANS_FAILED,

    } from "../actions/PlaidActions";

import {
    CLEAR_PLAID
} from "../actions/LogoutAction";

import {
    CATEGORY_UPDATE_SUCCESS,
    BLOCKS_CATEGORY_LOADING, 
    BLOCKS_CATEGORY_FAILED
} from "../actions/userBlocks";

import {
    ADD_DEFAULT_CATEGORIES_LOADING,
    ADD_DEFAULT_CATEGORIES_SUCCESS,
    ADD_DEFAULT_CATEGORIES_FAILED,
    SELECT_CATEGORIES_SUCCESS,
    ADD_MANUAL_BLOCKS_SUCCESS,
    EDIT_MANUAL_BLOCKS_SUCCESS
    } from "../actions/ManualActions";

export const reducer = (state = initialState, action) => {
    switch(action.type){
        case EDIT_MANUAL_BLOCKS_SUCCESS:
            return {
                ...state,
                categories:[...state.categories,action.payload],
                error: null
            };
        case ADD_MANUAL_BLOCKS_SUCCESS:
            return {
                ...state,
                categories:[...state.categories,action.payload],
                error: null
            };
        case SEND_LINK_TOKEN_LOADING:
            return {
                ...state,
                isFetching: true,
                error: null
            };
        case SEND_LINK_TOKEN_FAILED:
            return {
                ...state,
                isFetching: false,
                error: action.payload
            }
        case SEND_LINK_TOKEN_SUCCESS:
            return {
                ...state,
                isFetching: false,
                error: null
            }
        case GET_TRANS_LOADING:
            return {
                ...state,
                isFetching: true,
                error: null
            };
        case GET_TRANS_FAILED:
            return {
                ...state,
                isFetching: false,
                error: action.payload
            }
        case GET_TRANS_SUCCESS:
            return {
                ...state,
                isFetching: false,
                error: null,
                accounts:action.payload.accounts,
                categories:action.payload.Categories,
                transactions:sortTrans(action.payload.Categories) // function at bottom of file
            }
        case CLEAR_PLAID:
            return {
                ...initialState
            }
        case CATEGORY_UPDATE_SUCCESS:
            return {
                ...state,
                categories:updateCategory(state.categories, action.payload.categoryid, action.payload.amount) // function at bottom of file
            }
        case ADD_DEFAULT_CATEGORIES_LOADING:
            return {
                ...state,
                isFetching: true,
                error: null
            };
        case ADD_DEFAULT_CATEGORIES_FAILED:
            return {
                ...state,
                isFetching: false,
                error: action.payload
            }
        case ADD_DEFAULT_CATEGORIES_SUCCESS:
            return {
                ...state,
                isFetching: false,
                error: null,
                categories:action.payload
            }
        case SELECT_CATEGORIES_SUCCESS:
            return {
                ...state,
                categories:action.payload
            }
        case BLOCKS_CATEGORY_LOADING:
            return {
                ...state,
                isFetching: true,
                error: null
            }
        case BLOCKS_CATEGORY_FAILED:
            return {
                ...state,
                isFetching: false,
                error: action.payload
            }
        default:
            return state;
    }
}

const initialState = {
    error:null,
    isFetching:false,
    transactions:[],
    Balance:0.00,
    categories:[],
    accounts:[],
    
    
};

function sortTrans(cats){
    // takes all transactions and puts them into one array. sorts them Newest to oldest
    // Each category object has an array for their transactions
    let transArr = [];
    // creates array of transaction arrays
    const catTransArr = cats.map((cat)=>{
        return cat.transactions.map((trans)=>{
            return {...trans, category:cat.name}
        })
    });
    // maps through all arrays of transactions
    // takes each transaction and puts them into one array
    catTransArr.forEach(trans => {
        transArr = [...transArr, ...trans];
    });
    // sorts array from newest to oldest
    transArr = transArr.sort((a,b)=>{
        return new Date(b.payment_date) - new Date(a.payment_date);
    })

    return transArr;
}

function updateCategory(arr,categoryid, amount){
    // maps through array and updates the category that was updated 
    const newCategory = arr.map(c =>
        c.id === categoryid
          ? { ...c, budget: amount }
          : c
      );
    
    return newCategory;
}