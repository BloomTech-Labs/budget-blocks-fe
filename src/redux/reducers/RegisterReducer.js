
import {
    REGISTER_USER_LOADING,
    REGISTER_USER_FAILED,
    REGISTER_USER_SUCCESS
    } from "../actions/RegisterActions"
export const reducer = (state = initialState, action) => {
    switch(action.type){
        case REGISTER_USER_LOADING:
            return {
                ...state,
                isFetching: true,
                error: null
            };
        case REGISTER_USER_FAILED:
            return {
                ...state,
                isFetching: false,
                error: action.payload
            }
        case REGISTER_USER_SUCCESS:
            return {
                ...state,
                isFetching: false,
                error: null
            }
        default:
            return state;
    }
}

const initialState = {
    error:null,
    isFetching:false,

};