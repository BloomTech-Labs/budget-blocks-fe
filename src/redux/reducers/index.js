import { combineReducers } from 'redux';
import { transactionReducer } from './transactionReducer';
import { onBoardReducer } from './onBoardReducer';
import { userReducer } from './userReducer';

export default combineReducers({
  trans: transactionReducer,
  onBoard: onBoardReducer,
  users: userReducer,
});
