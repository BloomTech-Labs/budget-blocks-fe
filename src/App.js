import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { Security, SecureRoute, LoginCallback } from '@okta/okta-react';
import { Container } from 'semantic-ui-react';

// SECTION Components
import config from './components/okta/config';
import Home from './components/Home';
import Dashboard from './components/Dashboard';
import Clickthrough from './components/Clickthrough';

// SECTION Onboarding Components
import Onboarding from './components/onboarding/OnBoarding';
import Welcome from './components/onboarding/Welcome';
import AccountInfo from './components/onboarding/AccountInfo';
import BankLink from './components/onboarding/BankLink';
import BudgetPreview from './components/onboarding/BudgetPreview';
import BudgetView from './components/onboarding/BudgetView';
import Income from './components/onboarding/Income';
import AnalyzerDashboard from './components/onboarding/dashboard/AnalyzerDashboard'
import BudgetCategory from './components/onboarding/BudgetCatergory'

// SECTION Goals Components
import HomeGoal from './components/onboarding/goals/HouseGoal'
import FoodGoal from './components/onboarding/goals/FoodGoal'
import TransportGoal from './components/onboarding/goals/TransportGoal'
import PersonalGoal from './components/onboarding/goals/PersonalGoal'

//SECTION CSS
import './App.css';

const App = () => (
  <Router>
    <Security {...config.oidc}>
      <Container>
        <SecureRoute path="/dashboard" component={Dashboard} />
        <SecureRoute path="/onboarding" exact component={Onboarding} />
        <Route path="/" exact component={Home} />
        <Route path="/clickthrough" component={Clickthrough} />
        <Route path="/implicit/callback" component={LoginCallback} />
        <Route path="/onboarding/welcome" component={Welcome} />
        <Route path="/onboarding/accountinfo" component={AccountInfo} />
        <Route path="/onboarding/banklink" exact component={BankLink} />
        <Route
          path="/onboarding/budgetpreview"
          exact
          component={BudgetPreview}
        />
        <Route path="/onboarding/budgetview" exact component={BudgetView} />
        <Route path="/onboarding/income" exact component={Income} />
        <Route path="/onboarding/analyzerview" exact component={AnalyzerDashboard} />
        <Route path="/onboarding/budgetcategory" exact component={BudgetCategory} />
        <Route path="/goal/house" exact component={HomeGoal} />
        <Route path="/goal/food" exact component={FoodGoal} />
        <Route path="/goal/transport" exact component={TransportGoal} />
        <Route path="/goal/personal" exact component={PersonalGoal} />
      </Container>
    </Security>
  </Router>
);

export default App;
