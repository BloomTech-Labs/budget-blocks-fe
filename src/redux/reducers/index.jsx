import { combineReducers } from 'redux';
import { DashboardReducer } from './DashboardReducer';
import { onBoardReducer } from './onBoardReducer';

export default combineReducers({
    dash: DashboardReducer,
    onBoard: onBoardReducer
})