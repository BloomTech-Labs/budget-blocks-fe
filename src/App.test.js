import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, render as enzymeRender } from 'enzyme';
import { applyMiddleware, createStore, combineReducers } from "redux";
import thunk from "redux-thunk";
import logger from "redux-logger";
import { Provider } from "react-redux";
import { reducer as plaidReducer } from "./redux/reducers/PlaidReducer";
import { reducer as registerReducer } from "./redux/reducers/RegisterReducer";
<<<<<<< HEAD
// import { reducer as profileReducer } from "./redux/reducers/ProfileReducer";
=======
>>>>>>> 076a0639fdefbe992ef4976c77819058df844587
import { BrowserRouter as Router } from "react-router-dom";

const initialState = {
  user: {
    id: null,
    token: "",
    message: "",
    LinkedAccount: true
  },
  error: null,
  isFetching: false
};

let loginReducer = (state = initialState, action) => {
  switch (action.type) {
    default:
      return state;
  }
}


<<<<<<< HEAD
const store = createStore(combineReducers({loginReducer,plaidReducer,registerReducer}), applyMiddleware(thunk, logger));
=======
const store = createStore(combineReducers({ loginReducer, plaidReducer, registerReducer }), applyMiddleware(thunk, logger));
>>>>>>> 076a0639fdefbe992ef4976c77819058df844587

configure({ adapter: new Adapter() });

test('App renders correctly', () => {
  expect(render(
    <Provider store={store}>
      <Router>
        <App />
      </Router>
    </Provider>
  )).toMatchSnapshot();
});
