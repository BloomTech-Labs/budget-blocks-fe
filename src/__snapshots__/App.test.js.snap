// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="nav-bar"
        >
          <div
            class="nav-logo"
          >
            <img
              class="image"
              src="Logo_113x83.svg"
            />
            <img
              class="heading"
              src="budget_blocks.png"
            />
          </div>
          <div
            class="nav-action"
          >
            <a
              href="/login"
            >
              Log Out
            </a>
          </div>
        </div>
        <div
          class="App"
        >
          <div
            class="home"
          >
            <div
              class="MuiContainer-root MuiContainer-maxWidthSm"
            >
              <div
                style="background-color: rgb(255, 255, 255); height: 100vh;"
              >
                <img
                  alt="logo"
                  class="App-logo"
                  src="Logo.svg"
                />
                <img
                  alt="budget_blocks"
                  class="budget_img"
                  src="budget_blocks.png"
                />
                <div
                  class="buttons"
                >
                  <a
                    class="links"
                    href="/register"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-outlined signup"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Sign Up
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </a>
                  <a
                    class="links"
                    href="/login"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-outlined signin"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Sign In
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="nav-bar"
      >
        <div
          class="nav-logo"
        >
          <img
            class="image"
            src="Logo_113x83.svg"
          />
          <img
            class="heading"
            src="budget_blocks.png"
          />
        </div>
        <div
          class="nav-action"
        >
          <a
            href="/login"
          >
            Log Out
          </a>
        </div>
      </div>
      <div
        class="App"
      >
        <div
          class="home"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthSm"
          >
            <div
              style="background-color: rgb(255, 255, 255); height: 100vh;"
            >
              <img
                alt="logo"
                class="App-logo"
                src="Logo.svg"
              />
              <img
                alt="budget_blocks"
                class="budget_img"
                src="budget_blocks.png"
              />
              <div
                class="buttons"
              >
                <a
                  class="links"
                  href="/register"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined signup"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Sign Up
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </a>
                <a
                  class="links"
                  href="/login"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined signin"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Sign In
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
